### ISSUE TITLE
Move param handlers to core part

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#1912]
## 🚀 Feature

As discussed in https://github.com/Project-MONAI/MONAI/pull/1949#issuecomment-815283287 moving https://github.com/pytorch/ignite/blob/master/ignite/contrib/handlers/param_scheduler.py to the core part: https://github.com/pytorch/ignite/tree/master/ignite/handlers could be great addition for users, apps and toolkits which do not use contrib module (e.g. MONAI).

While moving the code, we would like to ensure BC-compatibilty for at least one future release. Once the code has been moved we can put deprecation message for classes and methods inside contrib module. `contrib/handlers/param_scheduler.py` will be removed in v0.6.0.

A follow-up feature to this feature request could be enlarging the scope of `ParamScheduler` to scheduling any parameter (e.g. hyperparameters) like it is done in MONAI: https://github.com/Project-MONAI/MONAI/pull/1949

cc @danieltudosiu

**Comment by danieltudosiu** [comment]
@vfdev-5 thanks for ccing.

Just to make sure I get your right. 

1) For now we will just move ParamScheduler out of contrib and add MONAI ParameterScheduler to Ignite?

2) In this issue we will not aim to combine the two into a better class because we want to maintain backwards compatibility between 0.5 and 0.4. Right? 

Thanks!

**Comment by vfdev-5** [comment]
@danieltudosiu yes, the point (2) is correct. For this issue we want only move the code and keep backward compatibility between v0.4.4 (now) up to future release v0.6.0.

In another PR associated to #1913, we would like to extend moved to core ParamScheduler to any parameter scheduler still keeping contrib API backward compatible.

Once we are done with both features and we released a new version, then I think we can adapt/extend MONAI's codebase if maintainers are OK with that.

**Comment by danieltudosiu** [comment]
@vfdev-5 perfect, will aim to have it done by the end of the weekend.

**Comment by vfdev-5** [comment]
Sounds great, thanks @danieltudosiu !

**Comment by vfdev-5** [comment]
Hi @danieltudosiu how things are going with this issue ?

**Comment by danieltudosiu** [comment]
@vfdev-5 sorry I had a hectic week with family issues and PhD. This week I will jump on it.

**Comment by vfdev-5** [comment]
@fco-dv please start with this issue instead of #1913 while @danieltudosiu could directly start with #1913

**Comment by fco-dv** [comment]
Yes sure @vfdev-5
